src/hooks/useActiveDescendantListFocus/index.js

```diff
@@ -42,6 +43,45 @@ function useActiveDescendantListFocus({
   const backArrow = isHorizontalList ? ARROW_DIRECTIONS.LEFT : ARROW_DIRECTIONS.UP;
 
   const [visualFocusItemIndex, setVisualFocusItemIndex] = useState(-1);
+  const visualFocusItemId = itemsIds[visualFocusItemIndex];
+  const previousItemIds = usePrevious(itemsIds);
+  const previousVisualFocusItemIndex = usePrevious(visualFocusItemIndex);
+  const prevVisualFocusItemId = usePrevious(itemsIds[visualFocusItemIndex]);
+
+  const getFirstSelectable = useCallback(() => {
+    for (let idx = 0; idx < itemsIds.length; idx++) {
+      if (isItemSelectable(idx)) {
+        return idx;
+      }
+    }
+    return -1;
+  }, [itemsIds]);
+
+  useEffect(() => {
```

Comment:
it's a very scary use effect for outside reader... it's take a few moments to understand what is purpose.

If I were you, I would move all the logic which related to keep the id with the same value when the list change to a different custom hook "useKeepSameItemFocusedWhenItemsListChange" including all the use previous usage and stuff... you don't need to do it, I will do it before merging.